### PR TITLE
CORE - 3.4.1.4 Processing Rules (minus Subject comparison)

### PR DESCRIPTION
**Pull request by brjeter**
#### What does this PR do (if it's not clear from the Title)? Please include any specification references that apply.
- ```The responder MUST ultimately reply to an <AuthnRequest> with a <Response> message containing one or more assertions that meet the specifications defined by the request```
- ```At least one statement in at least one assertion MUST be a <saml:AuthnStatement> that describes the authentication performed by the responder or authentication service associated with it.```
- ```The resulting assertion(s) MUST contain a <saml:AudienceRestriction> element referencing the requester as an acceptable relying party. Other audiences MAY be included as deemed appropriate by the identity provider.```

#### Checklist:
- [ ] SAML Spec Table of Contents documentation updated

### CODE REVIEW

File: ctk/common/src/main/kotlin/org/codice/compliance/verification/core/responses/CoreAuthnRequestProtocolVerifier.kt [hunk 1]
```diff
@@ -38,11 +42,26 @@ class CoreAuthnRequestProtocolVerifier(override val response: Node,
...
+                    node = response)
+
+        if (assertions.all { it.children("AuthnStatement").isEmpty() })
+            throw SAMLComplianceException.create(SAMLCore_3_4, SAMLCore_3_4_1_4b,
                     message = "AuthnStatement not found in any of the Assertions.",
                     node = response)
+
+        if (assertions.any {
+                    it.recursiveChildren("AudienceRestriction").flatMap { it.children("Audience") }
+                            .none { it.textContent == "https://samlhost:8993/services/saml" }
```
Comment (blen-desta) [inline ctk/common/src/main/kotlin/org/codice/compliance/verification/core/responses/CoreAuthnRequestProtocolVerifier.kt]: ✏️ `SP_ISSUER` in `TestCommon` should be `https://samlhost:8993/services/saml`. I suggest using that instead.

**Comment by coyotesqrl** [comment]
❓ Does this complete Core 3.4.1.4?

File: ctk/common/src/main/kotlin/org/codice/compliance/verification/core/responses/CoreAuthnRequestProtocolVerifier.kt (hunk 1 above)
Comment (coyotesqrl) [reply ctk/common/src/main/kotlin/org/codice/compliance/verification/core/responses/CoreAuthnRequestProtocolVerifier.kt]: ☝️ Definitely in favor or removing this magic string.

Comment (brjeter) [reply ctk/common/src/main/kotlin/org/codice/compliance/verification/core/responses/CoreAuthnRequestProtocolVerifier.kt]: Sorry, I used to do magic-oriented programming before I started going object-oriented. I slip back into that occasionally 😉

**Comment by brjeter** [comment]
This does not complete 3.4.1.4. There are 2 MUSTs left that depend on @bakejeyner's subject comparison PR.

**Comment by brjeter** [comment]
build now

**Pull request merged**